src/Flight/FlightService.java

```diff
@@ -1,6 +1,143 @@
 package Flight;
 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
 // TODO: Implement the Service Layer, this contains all Business Logic (anything that makes sense in the scope of a business)
 public class FlightService {
+    private Flight[] flights = new Flight[5];
+
+
+    public Flight[] getFlightInfo(){
+        if (isEmpty(flights)) {
+            System.out.println("No flight info");
+            return null;
+        } else {
+            System.out.println("Returning flight information....");
+            return flights;
+        }
+    }
+
+    public void createFlight(Flight flight){
+        int indexToReplace = getFirstNull(flights);
+        if (indexToReplace == -1) {
+            System.out.println("Sorry, our flight database is full, please try again later");
+        } else if(!validateMinFlight(flight)){
+            System.out.println("Invalid information provided.");
+        } else {
+            flights[indexToReplace] = flight;
+            System.out.printf("Flight %s successfully added\n", flight);
+        }
+    }
+
+    public Flight findById(int flightNumber){
+        for (Flight flight : flights) {
+            if (flight != null && flight.flightNumber == flightNumber) {
+                return flight;
+            }
+        }
+
+        return null;
+    }
+
+
+    public boolean updateFlight(Flight flightToUpdate, String timeArrival, String timeDeparture, int pilot, int airline) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
```

Comment:
Is line 46, how we are dealing with input validation for the date and time that the user will enter?